app/src/tx_reviewer.rs

```diff
@@ -0,0 +1,452 @@
+            _ => "devnet",
+        };
+
+        let fields = [Field {
+            name: "Network",
+            value: network_type,
+        }];
+        review(&fields, "Review Network")
+    }
+
+    pub fn review_gas_amount(gas_amount: &I32) -> Result<(), ErrorCode> {
+        let mut output = [0; 11];
+        let num_str_bytes = gas_amount.to_str(&mut output);
+        if num_str_bytes.is_none() {
+            return Err(ErrorCode::Overflow);
+        }
+        let value = bytes_to_string(&num_str_bytes.unwrap())?;
+        let fields = [Field {
+            name: "GasAmount",
+            value,
+        }];
+        review(&fields, "Review Gas Amount")
+    }
+
+    pub fn review_gas_price(&mut self, gas_price: &U256) -> Result<(), ErrorCode> {
+        let from_index = self.index;
+        let to_index = self.write_alph_amount(gas_price)?;
+        let value = self.get_str_from_range((from_index, to_index))?;
+        let fields = [Field {
+            name: "GasPrice",
+            value,
+        }];
+        review(&fields, "Review Gas Price")
+    }
+
+    pub fn review_input(
+        &mut self,
+        input: &TxInput,
+        current_index: usize,
+        temp_data: &[u8],
+    ) -> Result<(), ErrorCode> {
+        let InputIndexes {
+            review_message,
+            address,
+        } = self.prepare_input(input, current_index, temp_data)?;
+        let review_message = self.get_str_from_range(review_message)?;
+        let address = self.get_str_from_range(address)?;
+        let fields = [Field {
+            name: "Address",
+            value: address,
+        }];
+        review(&fields, review_message)
+    }
+
+    pub fn review_output(
+        &mut self,
+        output: &AssetOutput,
+        current_index: usize,
+        temp_data: &[u8],
+    ) -> Result<(), ErrorCode> {
+        let OutputIndexes {
+            review_message,
+            alph_amount,
+            address,
+            token,
+        } = self.prepare_output(output, current_index, temp_data)?;
+        let review_message = self.get_str_from_range(review_message)?;
+        let alph_amount = self.get_str_from_range(alph_amount)?;
+        let address = self.get_str_from_range(address)?;
+        let address_field = Field {
+            name: "Address",
+            value: address,
+        };
+        let alph_amount_field = Field {
+            name: "ALPH",
+            value: alph_amount,
+        };
+        if token.is_none() {
+            let fields = [address_field, alph_amount_field];
+            return review(&fields, review_message);
+        }
+
+        let TokenIndexes {
+            token_id,
+            token_amount,
+        } = token.unwrap();
+        let token_id = self.get_str_from_range(token_id)?;
+        let token_amount = self.get_str_from_range(token_amount)?;
+        let fields = [
+            address_field,
+            alph_amount_field,
+            Field {
+                name: "TokenId",
```

Comment:
I tested ETH, but for transfer tokens tx, ETH does not display token info because contract calls require blind signing. But in ETH the gas fee is shown as `Max Fees`, so perhaps `Token ID `would be better.

Reply:
Yes, let's use `Token ID` and `Token Amount`